To-do list/Controllers/TaskDetailsViewController.swift

```diff
@@ -7,39 +7,44 @@
 
 import UIKit
 
+protocol TaskDetailsViewControllerDelegate: AnyObject {
+    func sendTaskDetails(_: TaskDetailsViewController, didCreateUpdate task: Task)
+}
+
 class TaskDetailsViewController: UIViewController {
     
+    weak var delegate: TaskDetailsViewControllerDelegate?
+    
     @IBOutlet weak var titleTextField: UITextField!
     @IBOutlet weak var descriptionTextField: UITextField!
     @IBOutlet weak var statusButtonOutlet: UIButton!
     
-    var users = [User]()
-    var userIndex = Int()
-    var taskIndex = Int()
+    
+
+    var task = Task()
     var counter = 0
-    var status = "to do"
-    var titleField = ""
+    var status = Task.Status.todo
     var choosenTableViewCell = Bool()
```

Comment:
1. this is very bad name for variable. 
2. You don't need this variable at all
3. Instead, controller should have enum `ViewType` with 2 values: `createTask, updateTask`. createTask - assigned if TaskDetailsViewController is used to create new task.

Reply:
variable deleted, enum created